base/src/test/resources/success/common/src/Arith/Nat/Core.aya

```diff
@@ -60,3 +63,10 @@ private overlap def *-assoc-lemma : Associative (*)
 | 0, _, _ => idp
 | suc m, _, _ => <=3=> (+-*-distl _) (pmap (+ _) *-assoc) +-comm
 inline def *-assoc {m n k : Nat} => *-assoc-lemma m n k
+
+overlap def infix =? (m n : Nat) : Bool
```

Comment:
This definition is not overlapping. I'll try to write a warning that detects such non-overlapping `overlap` definitions.